WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactable.cs

```diff
@@ -1,50 +1,93 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using System;
 
-public class Interactable : MonoBehaviour {
+public class InteractedWithEventArgs : EventArgs
+{
+    public InteractionType InteractionType;
+}
 
-    public static Interactable currentInteractable;
+public enum InteractionType { Sniff, Use, Attack}
 
-    protected Color _interactableColor = new Color(1f, 1f, 1f, .25f);
-    protected Color _defaultColor = new Color(1f, 1f, 1f, 0f);
-    protected Color _interactionColor = new Color(.5f, .8f, .2f, .25f);
+public class Interactable : MonoBehaviour {
 
-    public void SetAsCurrentInteractable()
+    private static Interactable currentInteractable_internal;
+    public static Interactable CurrentInteractable
     {
-        if (currentInteractable != this)
+        set
```

Comment:
Yeah. it's just a setter function associated with that value. So that when you call Interactable.CurrentInteractable = [value] that set function gets run on [value]. It's very useful when you want to ensure some processing is done every time you set a value, while allowing you to set values without method calls.

Reply:
Thanks!